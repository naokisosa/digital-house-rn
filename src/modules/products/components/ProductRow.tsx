import { StyleSheet, Text, View } from 'react-native';
import React from 'react';

const ProductRow = () => {
  return (
    <View>
      <Text>ProductRow</Text>
    </View>
  );
};

export default ProductRow;

const styles = StyleSheet.create({});
