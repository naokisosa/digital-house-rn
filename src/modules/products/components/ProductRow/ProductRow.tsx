import { Pressable, StyleSheet, View } from 'react-native';
import React, { memo } from 'react';
import { Product } from '@services/types';
import { BaseText, Image } from '@components';
import { colors } from '@style';
import { formatDate } from '@utils';

interface Props {
  product?: Product;
  onPress: () => void;
}

const ProductRow = ({ product, onPress }: Props) => {
  const symbol = product?.is_redemption ? '+' : '-';

  return (
    <Pressable style={styles.container} onPress={onPress}>
      <Image source={{ uri: product?.image }} style={styles.image} size="small" />
      <View style={styles.textContainer}>
        <BaseText style={styles.name}>{product?.product}</BaseText>
        <BaseText style={styles.createdAt}>{formatDate(product?.createdAt)}</BaseText>
      </View>
      <View>
        <BaseText style={styles.points}>
          <BaseText style={styles.symbol(product?.is_redemption)}>{symbol}</BaseText>
          {product?.points}
        </BaseText>
      </View>

      <BaseText style={styles.icon}>{'>'}</BaseText>
    </Pressable>
  );
};

export default memo(ProductRow);

const styles = StyleSheet.create<any>({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    marginVertical: 4,
  },
  image: {
    borderRadius: 10,
  },
  textContainer: {
    marginLeft: 11,
    flex: 1,
  },
  name: {
    fontWeight: '800',
    fontSize: 14,
    lineHeight: 19.12,
    marginBottom: 7,
  },
  createdAt: {
    fontWeight: '400',
    fontSize: 12,
    lineHeight: 16.39,
  },
  points: {
    fontWeight: '800',
    fontSize: 16,
    lineHeight: 21.86,
  },
  symbol: (isRedemption: boolean) => ({
    color: isRedemption ? colors.green : colors.red,
  }),
  icon: {
    fontSize: 13,
    fontWeight: '800',
    marginLeft: 17,
  },
});
