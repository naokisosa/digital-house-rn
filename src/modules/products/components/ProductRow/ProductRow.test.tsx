import 'react-native';
import React from 'react';

import ProductRow from './ProductRow';
import renderer from 'react-test-renderer';
import { SINGLE_PRODUCT } from '@mocks';

describe('Test ProductRow component', () => {
  it('renders correctly with mock product', () => {
    const { toJSON } = renderer.create(<ProductRow product={SINGLE_PRODUCT} onPress={jest.fn} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
