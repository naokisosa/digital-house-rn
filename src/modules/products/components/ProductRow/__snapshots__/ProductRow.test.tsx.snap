// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProductRow component renders correctly with mock product 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "marginVertical": 4,
    }
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        [
          {
            "height": 55,
            "width": 55,
          },
          {
            "borderRadius": 10,
          },
        ],
      ]
    }
  >
    <FastImageView
      defaultSource={
        {
          "testUri": "../../../src/assets/placeholder.png",
        }
      }
      resizeMode="cover"
      source={
        {
          "uri": "https://loremflickr.com/640/480/city",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      {
        "flex": 1,
        "marginLeft": 11,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Avenir",
          },
          {
            "fontSize": 14,
            "fontWeight": "800",
            "lineHeight": 19.12,
            "marginBottom": 7,
          },
        ]
      }
    >
      Recycled Wooden Salad
    </Text>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Avenir",
          },
          {
            "fontSize": 12,
            "fontWeight": "400",
            "lineHeight": 16.39,
          },
        ]
      }
    >
      9 de diciembre, 2022
    </Text>
  </View>
  <View>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Avenir",
          },
          {
            "fontSize": 16,
            "fontWeight": "800",
            "lineHeight": 21.86,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "Avenir",
            },
            {
              "color": "#FF0000",
            },
          ]
        }
      >
        -
      </Text>
      44871
    </Text>
  </View>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontFamily": "Avenir",
        },
        {
          "fontSize": 13,
          "fontWeight": "800",
          "marginLeft": 17,
        },
      ]
    }
  >
    &gt;
  </Text>
</View>
`;
